services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -456,5 +489,12 @@ public int hashCode() {
         return result;
     }
 
+    private static void checkAndValidateClass(Class<?> type, boolean isPut) {
+        Validate.paramNotNull(type, "type");
+        Validate.isTrue(!type.isAssignableFrom(List.class),
+                        String.format(VALIDATE_TYPE_ERROR, List.class.getSimpleName(), isPut ? "put" : "get", "List"));
```

Comment:
`getSimpleName` is actually very expensive, can we just use literal `List`?

Reply:
Changed